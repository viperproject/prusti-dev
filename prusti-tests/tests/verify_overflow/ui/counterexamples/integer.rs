// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true

#![allow(unused)]

use prusti_contracts::*;

#[requires(a == Int::new(2))] //force specific counterexample
#[ensures(a == Int::new(5))]
fn test1(a: Int) {}

#[requires(a == Int::new(10))] //force specific counterexample
#[ensures(!result)]
fn test2(a: Int, b: Int) -> bool{
    let c = a + b;
    c == Int::new(30)
}


#[requires(a == Int::new(10) && c == Int::new(11) && b == Int::new(0))] //force specific counterexample
#[ensures(result)]
fn test3(a: Int, b: Int, c: Int) -> bool {
    a + c >= b + c
}

fn main() {}
