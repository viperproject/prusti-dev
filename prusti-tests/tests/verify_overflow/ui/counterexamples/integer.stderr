error: [Prusti: verification error] postcondition might not hold.
 --> $DIR/integer.rs:8:11
  |
8 | #[ensures(a == Int::new(5))]
  |           ^^^^^^^^^^^^^^^^
  |
note: the error originates here
 --> $DIR/integer.rs:9:1
  |
9 | fn test1(a: Int) {}
  | ^^^^^^^^^^^^^^^^^^^
note: counterexample for "a"
       value:   2
 --> $DIR/integer.rs:9:10
  |
9 | fn test1(a: Int) {}
  |          ^
note: counterexample for "result"
       value:   ()
 --> $DIR/integer.rs:9:18
  |
9 | fn test1(a: Int) {}
  |                  ^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/integer.rs:12:11
   |
12 | #[ensures(!result)]
   |           ^^^^^^^
   |
note: the error originates here
  --> $DIR/integer.rs:13:1
   |
13 | / fn test2(a: Int, b: Int) -> bool{
14 | |     let c = a + b;
15 | |     c == Int::new(30)
16 | | }
   | |_^
note: counterexample for "a"
       value:   10
  --> $DIR/integer.rs:13:10
   |
13 | fn test2(a: Int, b: Int) -> bool{
   |          ^
note: counterexample for "b"
       value:   20
  --> $DIR/integer.rs:13:18
   |
13 | fn test2(a: Int, b: Int) -> bool{
   |                  ^
note: counterexample for "c"
       value:   30
  --> $DIR/integer.rs:14:13
   |
14 |     let c = a + b;
   |             ^^^^^
note: counterexample for "c"
       value:   30
  --> $DIR/integer.rs:15:5
   |
15 |     c == Int::new(30)
   |     ^
note: counterexample for "result"
       value:   true
  --> $DIR/integer.rs:15:5
   |
15 |     c == Int::new(30)
   |     ^^^^^^^^^^^^^^^^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/integer.rs:21:37
   |
21 | fn test3(a: Int, b: Int, c: Int) -> bool {
   |                                     ^^^^
   |
note: the error originates here
  --> $DIR/integer.rs:21:1
   |
21 | / fn test3(a: Int, b: Int, c: Int) -> bool {
22 | |     a + c >= b + c
23 | | }
   | |_^
note: counterexample for "a"
       value:   10
  --> $DIR/integer.rs:21:10
   |
21 | fn test3(a: Int, b: Int, c: Int) -> bool {
   |          ^
note: counterexample for "b"
       value:   0
  --> $DIR/integer.rs:21:18
   |
21 | fn test3(a: Int, b: Int, c: Int) -> bool {
   |                  ^
note: counterexample for "c"
       value:   11
  --> $DIR/integer.rs:21:26
   |
21 | fn test3(a: Int, b: Int, c: Int) -> bool {
   |                          ^
note: counterexample for "result"
       value:   false
  --> $DIR/integer.rs:22:5
   |
22 |     a + c >= b + c
   |     ^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

